In [1]:
#import dependencies

import numpy as np
import pandas as pd

In [2]:
#import the csv for analysis 
#reading it in as a dataframe 
# getting the head to examine the columns and rows

csv_path = "Resources/purchase_data.csv"
df = pd.read_csv(csv_path)
df.head()

Purchase ID             SN  Age Gender  Item ID  \
0            0        Lisim78   20   Male      108   
1            1    Lisovynya38   40   Male      143   
2            2     Ithergue48   24   Male       92   
3            3  Chamassasya86   24   Male      100   
4            4      Iskosia90   23   Male      131   

                                   Item Name  Price  
0  Extraction, Quickblade Of Trembling Hands   3.53  
1                          Frenzied Scimitar   1.56  
2                               Final Critic   4.88  
3                                Blindscythe   3.27  
4                                       Fury   1.44

In [3]:
# Getting a list of the unique player names. 

unique_players = df["SN"].unique()

## Player Count

In [4]:
# Counting the total number of players in the list of unique names for total player count

total_players = len(unique_players)
total_players = pd.DataFrame({"Total Players": [total_players]})
total_players

Total Players
0            576

In [5]:
# Getting a list of the unique items by item name

unique_items = df["Item Name"].unique()


In [6]:
# Total number of unique items
total_unique_items = len(unique_items)

In [7]:
#Average Purchase Price

avg_purchase_price = df["Price"].mean()

In [8]:
# Total Purchases By Counting the Number of Purchase IDs

total_purchases = df["Purchase ID"].count()
total_purchases

780

In [9]:
# Total Revenue Retrieved by summing the purchase price column

total_revenue = df["Price"].sum()
total_revenue

2379.77

## Purchasing Analysis (Total)

In [10]:
#Putting Together Purchasing Analysis Dataframe
purchasing_df = {
          "Number of Unique Items": [total_unique_items],
          "Average Purchase Price": [avg_purchase_price],
          "Total Number of Purchases": [total_purchases],
          "Total Revenue": [total_revenue]}
purchasing_df = pd.DataFrame(purchasing_df)

purchasing_df["Average Purchase Price"] = purchasing_df["Average Purchase Price"].map("${:,.2f}".format)
purchasing_df["Total Revenue"] = purchasing_df["Total Revenue"].map("${:,.2f}".format)

In [11]:
#Printing out Purchasing Analysis (Total)
purchasing_df

Number of Unique Items Average Purchase Price  Total Number of Purchases  \
0                     179                  $3.05                        780   

  Total Revenue  
0     $2,379.77

#  Gender Demographics

In [12]:
# Group data by Gender and getting the number of unique screen names

gender_demo = df.groupby("Gender")["SN"].nunique()

In [13]:
# Adding to a dataframe

gender_demo = pd.DataFrame({"Total Count": gender_demo})

#calculating percentages and adding to the dataframe

gender_percentages = (gender_demo["Total Count"] / gender_demo["Total Count"].sum() * 100)

gender_demo["Percentage of Players"] = gender_percentages

#formatting

gender_demo["Percentage of Players"] = gender_demo["Percentage of Players"].map("{:.2f}%".format)

In [14]:
gender_demo

Total Count Percentage of Players
Gender                                                  
Female                          81                14.06%
Male                           484                84.03%
Other / Non-Disclosed           11                 1.91%

# Purchase Analysis (Gender)

In [15]:
#Grab the count of purchase IDs for each gender

gender_purchase = df.groupby("Gender")["Purchase ID"].count()

#Create a new dataframe with the purchase count by gender
gender_purchase_df = pd.DataFrame({"Purchase Count": gender_purchase})

# Calculate the average purchase price for each gender

average_purchase = df.groupby("Gender")["Price"].mean()

# Map and format in a dataframe

gender_purchase_df["Average Purchase Price"] = average_purchase.map("${:.2f}".format)

# Calculate the total purchase price for each gender

total_purchase = df.groupby("Gender")["Price"].sum()

gender_purchase_df["Total Purchase Price"] = total_purchase.map("${:.2f}".format)

gender_purchase_df


Purchase Count Average Purchase Price  \
Gender                                                         
Female                            113                  $3.20   
Male                              652                  $3.02   
Other / Non-Disclosed              15                  $3.35   

                      Total Purchase Price  
Gender                                      
Female                             $361.94  
Male                              $1967.64  
Other / Non-Disclosed               $50.19

In [16]:
#Calculate the Average Purchase Total per Person by Gender

per_person_purchase = df.groupby("SN")["Price"].sum()
per_person_purchase_df = pd.DataFrame({"Total Purchase Per Person": per_person_purchase})

#per_person_purchase_df = pd.DataFrame({"Average Purchase Total Per Person By Gender": per_person_purchase})

# Merge dataframes on "SN" field to get the gender for each person

gender_merge_df = pd.merge(df, per_person_purchase_df, on="SN")

#Drop the duplicate screen names, keeping the first instance
gender_merge_df = gender_merge_df.drop_duplicates(subset="SN", keep="first") 

#Calcuage the average total purchase per person by gender

gender_per_person_purchase = gender_merge_df.groupby("Gender")["Total Purchase Per Person"].mean()

#Add the average to the gender purchase summary table

gender_purchase_df["Average Total Purchase Per Person"] = gender_per_person_purchase.map("${:.2f}".format)

gender_purchase_df



Purchase Count Average Purchase Price  \
Gender                                                         
Female                            113                  $3.20   
Male                              652                  $3.02   
Other / Non-Disclosed              15                  $3.35   

                      Total Purchase Price Average Total Purchase Per Person  
Gender                                                                        
Female                             $361.94                             $4.47  
Male                              $1967.64                             $4.07  
Other / Non-Disclosed               $50.19                             $4.56

## Age Demographics

In [17]:
# want to see the upper and lower levels of age in the original df

print(df["Age"].max())
print(df["Age"].min())

45
7


In [18]:
# Binning - creating the bins for the age ranges - max is 50 as 45 is the maximum age of a player

bins = [0, 9.90, 14.90, 19.90, 24.90, 29.90, 34.90, 39.90, 50]

#Assign names to bins 

names = ["Less than 10", "10-14", "15-19", "20-24", "25-29", "30-34", "35-39", "40 and Over"]

df["Age Range"] = pd.cut(df["Age"], bins, labels = names)

df

Purchase ID             SN  Age  Gender  Item ID  \
0              0        Lisim78   20    Male      108   
1              1    Lisovynya38   40    Male      143   
2              2     Ithergue48   24    Male       92   
3              3  Chamassasya86   24    Male      100   
4              4      Iskosia90   23    Male      131   
..           ...            ...  ...     ...      ...   
775          775     Aethedru70   21  Female       60   
776          776         Iral74   21    Male      164   
777          777     Yathecal72   20    Male       67   
778          778        Sisur91    7    Male       92   
779          779      Ennrian78   24    Male       50   

                                     Item Name  Price     Age Range  
0    Extraction, Quickblade Of Trembling Hands   3.53         20-24  
1                            Frenzied Scimitar   1.56   40 and Over  
2                                 Final Critic   4.88         20-24  
3                                  Blindscythe   3.27         20-24  
4                                         Fury   1.44         20-24  
..                                         ...    ...           ...  
775                                       Wolf   3.54         20-24  
776                           Exiled Doomblade   1.63         20-24  
777      Celeste, Incarnation of the Corrupted   3.46         20-24  
778                               Final Critic   4.19  Less than 10  
779                                       Dawn   4.60         20-24  

[780 rows x 8 columns]

In [19]:
#Group by age range

age_range = df.groupby("Age Range")

# Get the number of unique screen names by age range

age_range_player = df.groupby("Age Range")["SN"].nunique()

#create a dataframe with the total count of players by age range

age_range_player_df = pd.DataFrame({"Total Count": age_range_player})

# calculate the percentage of players by age range

age_range_percentage = (age_range_player_df["Total Count"] / age_range_player_df["Total Count"].sum()) * 100

#add a colum of the percentage of players by age range to the age_range df and format it
age_range_player_df["Percentage of Players"] = age_range_percentage.map("{:.2f}%".format)

age_range_player_df

Total Count Percentage of Players
Age Range                                      
Less than 10           17                 2.95%
10-14                  22                 3.82%
15-19                 107                18.58%
20-24                 258                44.79%
25-29                  77                13.37%
30-34                  52                 9.03%
35-39                  31                 5.38%
40 and Over            12                 2.08%

In [20]:
# Group the data by age range

age_range = df.groupby("Age Range")

# Get the total number of purchases by age range

age_range_purchases = age_range["Purchase ID"].count()

# Calculate average purchase price by age range

age_range_avg_price = age_range["Price"].mean()
age_range_avg_price = age_range_avg_price.map("${:.2f}".format)

# Calculate the total purchase value per age range

age_range_total_price = age_range["Price"].sum()
# can't format here 
# age_range_total_price = age_range_total_price.map("${:.2f}".format)

# put all of the above into a summary table

age_range_df = pd.DataFrame({"Purchase Count": age_range_purchases,
                            "Average Purchase Price": age_range_avg_price,
                            "Total Purchase Value": age_range_total_price})

age_range_df

Purchase Count Average Purchase Price  Total Purchase Value
Age Range                                                                
Less than 10              23                  $3.35                 77.13
10-14                     28                  $2.96                 82.78
15-19                    136                  $3.04                412.89
20-24                    365                  $3.05               1114.06
25-29                    101                  $2.90                293.00
30-34                     73                  $2.93                214.00
35-39                     41                  $3.60                147.67
40 and Over               13                  $2.94                 38.24

In [21]:
# Calcuate the average total purchases per person

age_range_avg_purchase_pp = age_range_df["Total Purchase Value"] / age_range_player_df["Total Count"]
age_range_avg_purchase_pp = age_range_avg_purchase_pp.map("${:.2f}".format)
total_purchase_value = age_range_df["Total Purchase Value"]
total_purchase_value = total_purchase_value.map("${:.2f}".format)

age_range_df["Average Purchase Total Per Person by Age Group"] = age_range_avg_purchase_pp
age_range_df["Total Purchase Value"] = total_purchase_value

age_range_df

Purchase Count Average Purchase Price Total Purchase Value  \
Age Range                                                                  
Less than 10              23                  $3.35               $77.13   
10-14                     28                  $2.96               $82.78   
15-19                    136                  $3.04              $412.89   
20-24                    365                  $3.05             $1114.06   
25-29                    101                  $2.90              $293.00   
30-34                     73                  $2.93              $214.00   
35-39                     41                  $3.60              $147.67   
40 and Over               13                  $2.94               $38.24   

             Average Purchase Total Per Person by Age Group  
Age Range                                                    
Less than 10                                          $4.54  
10-14                                                 $3.76  
15-19                                                 $3.86  
20-24                                                 $4.32  
25-29                                                 $3.81  
30-34                                                 $4.12  
35-39                                                 $4.76  
40 and Over                                           $3.19

## Top Spenders

In [22]:
# calculate the total purchase value per screen name/person

pp_total_purchase_value = df.groupby("SN")["Price"].sum()

# calculate the average purchase price per screen name/person

pp_avg_purchase_price = df.groupby("SN")["Price"].mean()

# Count the number of purchase IDs per screen name/person

pp_purchase_ids = df.groupby("SN")["Purchase ID"].count()

# put all of the above into a dataframe

pp_purchase_df = pd.DataFrame({"Purchase Count": pp_purchase_ids,
                              "Average Purchase Price": pp_avg_purchase_price,
                              "Total Purchase Value": pp_total_purchase_value})

# Sort the values by total purchase value in descending order / formatting
pp_purchase_df_sorted = pp_purchase_df.sort_values("Total Purchase Value", ascending=False)

pp_purchase_df_sorted["Average Purchase Price"] = pp_purchase_df_sorted["Average Purchase Price"].map("${:.2f}".format) 

pp_purchase_df_sorted["Total Purchase Value"] = pp_purchase_df_sorted["Total Purchase Value"].map("${:.2f}".format)

#print the top 5 spenders by total purchase value

pp_purchase_df_sorted.head(5)


Purchase Count Average Purchase Price Total Purchase Value
SN                                                                     
Lisosia93                 5                  $3.79               $18.96
Idastidru52               4                  $3.86               $15.45
Chamjask73                3                  $4.61               $13.83
Iral74                    4                  $3.40               $13.62
Iskadarya95               3                  $4.37               $13.10

## Most Popular Items

In [23]:
# Group the original dataframe by Item ID and Name

items = df.groupby(["Item ID", "Item Name"])

# Get the count of purchases per item

item_purchase = items["Purchase ID"].count()

# put into a dataframe

item_purchase_df = pd.DataFrame({"Purchase Count": item_purchase})

# merge new dataframe with original on Item Name

merge_price = pd.merge(df, item_purchase_df, on="Item Name")

merge_price = merge_price[["Item ID", "Item Name", "Price", "Purchase Count"]]
merge_price = merge_price.drop_duplicates(subset="Item Name", keep="first")
merge_price

# Group items by price

item_price = merge_price.groupby(["Item ID", "Item Name"])["Price"].sum()

# Calculate total price value by item id and name

total_item_price = items["Price"].sum()

item_purchase_df["Item Price"] = item_price

item_purchase_df["Total Purchase Value"] = total_item_price

sort_by_purchase = item_purchase_df.sort_values("Purchase Count", ascending=False)
sort_by_purchase["Item Price"] = sort_by_purchase["Item Price"].map("${:.2f}".format)
sort_by_purchase["Total Purchase Value"] = sort_by_purchase["Total Purchase Value"].map("${:.2f}".format)
sort_by_purchase.head(5)

Purchase Count  \
Item ID Item Name                                                      
92      Final Critic                                              13   
178     Oathbreaker, Last Hope of the Breaking Storm              12   
145     Fiery Glass Crusader                                       9   
132     Persuasion                                                 9   
108     Extraction, Quickblade Of Trembling Hands                  9   

                                                     Item Price  \
Item ID Item Name                                                 
92      Final Critic                                      $4.88   
178     Oathbreaker, Last Hope of the Breaking Storm      $4.23   
145     Fiery Glass Crusader                              $4.58   
132     Persuasion                                        $3.19   
108     Extraction, Quickblade Of Trembling Hands         $3.53   

                                                     Total Purchase Value  
Item ID Item Name                                                          
92      Final Critic                                               $59.99  
178     Oathbreaker, Last Hope of the Breaking Storm               $50.76  
145     Fiery Glass Crusader                                       $41.22  
132     Persuasion                                                 $28.99  
108     Extraction, Quickblade Of Trembling Hands                  $31.77

## Most Profitable Items

In [24]:
# Sort item purchase df by total purchase value

total_value_sort = item_purchase_df.sort_values("Total Purchase Value", ascending=False)

#format the columns

total_value_sort["Total Purchase Value"] = total_value_sort["Total Purchase Value"].map("${:.2f}".format)
total_value_sort["Item Price"] = total_value_sort["Item Price"].map("${:.2f}".format)

#Assign top 5 to a variable
total_value_top_five = total_value_sort.head(5)

# Print the top 5
total_value_top_five

Purchase Count  \
Item ID Item Name                                                      
92      Final Critic                                              13   
178     Oathbreaker, Last Hope of the Breaking Storm              12   
82      Nirvana                                                    9   
145     Fiery Glass Crusader                                       9   
103     Singed Scalpel                                             8   

                                                     Item Price  \
Item ID Item Name                                                 
92      Final Critic                                      $4.88   
178     Oathbreaker, Last Hope of the Breaking Storm      $4.23   
82      Nirvana                                           $4.90   
145     Fiery Glass Crusader                              $4.58   
103     Singed Scalpel                                    $4.35   

                                                     Total Purchase Value  
Item ID Item Name                                                          
92      Final Critic                                               $59.99  
178     Oathbreaker, Last Hope of the Breaking Storm               $50.76  
82      Nirvana                                                    $44.10  
145     Fiery Glass Crusader                                       $41.22  
103     Singed Scalpel                                             $34.80

## Three observable trends

1. While males made up the vast majority of players (84% of total players), female players (14.06% of total players) and players identifying as "other / non-disclosed" (1.91% of players) paid more on average for their video games ($3.20 and $3.35 respectively compared to the $3.02 males paid



2. People aged 20-24 purchased more video games than the other age groups (365 total purchase count) for a total purchase value of  1114.06 although they paid less on average ($2.93) per game than any other age group.


3. Although presumably an age group with more disposable income, people 40 and over bought the fewest games (13 total count), had the lowest total purchase value ($38.24) and spent the least per person, on average, per game ($3.19)